### PR TITLE
DWR-1219 Redshift schema changes to support drill down options by boolean data elements

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: reporting_olap/sql/V1_0_0_0__ddl.sql [hunk 1]
```diff
@@ -161,6 +161,16 @@ CREATE TABLE exam_claim_score_mapping (
 );
 
 -- dimensions
```
Comment (agorina) [inline reporting_olap/sql/V1_0_0_0__ddl.sql]: I could have done all these on a fly in the SQL query, but it fills better to include it into the schema.

File: reporting_olap/sql/V1_0_0_0__ddl.sql [hunk 2]
```diff
@@ -161,6 +161,16 @@ CREATE TABLE exam_claim_score_mapping (
...
+CREATE TABLE strict_boolean (
+  id smallint NOT NULL PRIMARY KEY SORTKEY,
+  code character varying(10) NOT NULL UNIQUE
+) DISTSTYLE ALL;
+
```
Comment (agorina) [inline reporting_olap/sql/V1_0_0_0__ddl.sql]: I could have done all these on a fly in the SQL query, but it fills better to include it into the schema.

File: reporting_olap/sql/V1_0_0_0__ddl.sql [hunk 3]
```diff
@@ -317,8 +327,13 @@ CREATE TABLE fact_student_exam (
...
   CONSTRAINT fk__fact_student_exam__school_year FOREIGN KEY(school_year) REFERENCES school_year(year),
   CONSTRAINT fk__fact_student_exam__school FOREIGN KEY(school_id) REFERENCES school(id),
-  CONSTRAINT fk__fact_student_exam__student FOREIGN KEY(student_id) REFERENCES student(id)
-)  COMPOUND SORTKEY (asmt_id, school_id, school_year, student_id);
+  CONSTRAINT fk__fact_student_exam__student FOREIGN KEY(student_id) REFERENCES student(id),
+  CONSTRAINT fk__fact_student_exam__iep FOREIGN KEY(iep) REFERENCES strict_boolean(id),
+  CONSTRAINT fk__fact_student_exam__lep FOREIGN KEY(lep) REFERENCES strict_boolean(id),
+  CONSTRAINT fk__fact_student_exam__section504 FOREIGN KEY(section504) REFERENCES boolean(id),
+  CONSTRAINT fk__fact_student_exam__economic_disadvantage FOREIGN KEY(economic_disadvantage) REFERENCES strict_boolean(id),
+  CONSTRAINT fk__fact_student_exam__migrant_statust FOREIGN KEY(migrant_status) REFERENCES boolean(id)
```
Comment (jtreuting) [inline reporting_olap/sql/V1_0_0_0__ddl.sql]: nit: looks like you have an extra "t" at the end of the constraint name

File: reporting_olap/sql/V1_0_0_0__ddl.sql [hunk 4]
```diff
@@ -317,8 +327,13 @@ CREATE TABLE fact_student_exam (
   CONSTRAINT fk__fact_student_exam__asmt FOREIGN KEY(asmt_id) REFERENCES asmt(id),
   CONSTRAINT fk__fact_student_exam__school_year FOREIGN KEY(school_year) REFERENCES school_year(year),
   CONSTRAINT fk__fact_student_exam__school FOREIGN KEY(school_id) REFERENCES school(id),
-  CONSTRAINT fk__fact_student_exam__student FOREIGN KEY(student_id) REFERENCES student(id)
-)  COMPOUND SORTKEY (asmt_id, school_id, school_year, student_id);
```
Comment (jtreuting) [inline reporting_olap/sql/V1_0_0_0__ddl.sql]: assuming you meant to get rid of this compound sortkey but jut checkin

File: reporting_olap/sql/V1_0_0_0__ddl.sql (hunk 3 above)
Comment (agorina) [reply reporting_olap/sql/V1_0_0_0__ddl.sql]: Thank you!

File: reporting_olap/sql/V1_0_0_0__ddl.sql (hunk 4 above)
Comment (agorina) [reply reporting_olap/sql/V1_0_0_0__ddl.sql]: Nope, total oversight. Thank you for noticing this!

**Pull request merged**